boa/src/builtins/number/mod.rs

```diff
@@ -254,13 +287,23 @@ impl Number {
         let this_num = Self::this_number_value(this, context)?;
         let precision = match args.get(0) {
             Some(n) => match n.to_integer(context)? as i32 {
-                x if x > 0 => n.to_integer(context)? as usize,
-                _ => 0,
+                0..=100 => n.to_integer(context)? as usize,
+                _ => {
+                    return Err(context.construct_range_error(
+                        "toFixed() digits argument must be between 0 and 100",
+                    ))
+                }
             },
             None => 0,
         };
-        let this_fixed_num = format!("{:.*}", precision, this_num);
-        Ok(JsValue::new(this_fixed_num))
+        // Get rid of the '-' sign for -0.0
```

Comment:
Step 6 from the spec seems to be missing here: `If x is not finite, return ! Number::toString(x).`
If you run `Infinity.toFixed()` the return is `"inf"` while it should be `"Infinity"`. I think there should be a 262 test for that, maybe we can get one added.

I would also suggest to add the spec as comments like in `parse_int`. That makes it very easy to read code and spec at the same time and see where and how they differ.

Reply:
Ah, you're right, I actually planned on implementing that since there is a test for it in `toExponential` implementation and I saw it in the spec. I must've forgotten, I'm sorry.